atlasdb-api/src/test/java/com/palantir/atlasdb/keyvalue/api/CandidateCellForSweepingTest.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.api;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Sets;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CandidateCellForSweepingTest {
+    private static final byte BYTE = (byte) 0xa;
+    private static final long TIMESTAMP = 1977;
+    private static final ImmutableSet<Integer> THREE_CELL_NAME_SIZES =
+            ImmutableSet.of(1, Cell.MAX_NAME_LENGTH / 2, Cell.MAX_NAME_LENGTH);
+    private static final ImmutableList<Cell> EXAMPLE_CELLS =
+            Sets.cartesianProduct(THREE_CELL_NAME_SIZES, THREE_CELL_NAME_SIZES).stream()
+                    .map(pair -> Cell.create(spawnBytes(pair.get(0)), spawnBytes(pair.get(1))))
+                    .collect(ImmutableList.toImmutableList());
+
+    private static final ImmutableList<Integer> SORTED_TIMESTAMPS_SIZES = ImmutableList.of(0, 1, 2, 100, 1000);
+
+    @Mock
+    private List<Long> MOCK_TIMESTAMPS;
+
+    @Test
+    public void candidateCellSizeWithLargerTimestampCollectionIsBigger() {
```

Comment:
I think your testing here is very thorough, but I think we can relax it a bit in favor of readability.

Reply:
It's also "best practice" to try and test a single concept per unit test. So for this, I'd just make a couple like:
`void candidateCellSizeHasCorrectSizeForOneTimestamp`, `void candidateCellSizeHasCorrectSizeForMultipleTimestamps`, `void candidateCellSizeIsEqualRegardlessOfLatestValueEmpty`